### ISSUE TITLE
[BUG] AD admin users were allowed to log in while the migration runs

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#42465]
**Rancher Server Setup**
- Rancher version:  `v2.7.4` upgraded to `v2.7.5` upgraded to v2.7.6-head `2cf6683 `
- Installation option (Docker install/Helm Chart): `Docker install`
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: `v1.26.7+k3s1`
- Cluster Type (Local/Downstream): Downstream Infrastructure provider k3s 
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: AD Admin
   
**Describe the bug**
Post upgrade from v2.7.4 to v2.7.5 to v2.7.6-head, AD admin user was able to log in while the un migration script runs. The admin user do not have any clusters listed - local/downstream. After a while, if we refresh the screen, admin user is logged out and after re logging in, we see all the clusters listed.

**To Reproduce**
1. On rancher v2.7.4, enable active directory
2. Create a downstream node driver k3s cluster 1 node all roles
3. Add a nesteduser, a non nested user from the AD server as cluster owner and cluster member respectively.
4. Add a few users from AD auth as project members.
5. Upgrade to rancher v2.7.5
6. Re login as the nested user and a few other users.
7. Upgrade rancher to v2.7.6-head
8. Login as the AD auth admin user while rancher starts and the migration script starts.

**Result**
AD auth admin user can log in but do not see any clusters listed and has an empty rancher server. Refresh the UI and re login, then the admin user can list all the clusters.

**Expected Result**
Expected the login to fail for admin user and that the admin user is not allowed to log in.

**Additional context**
<img width="400" alt="Screenshot2" src="https://github.com/rancher/rancher/assets/38144301/efa06903-bf5e-4559-ae6a-5c4510658d28">

**Comment by crobby** [comment]
@anupama2501 At the moment you manage to login this way, can you tell me what the following command is (ran against local cluster)
`kubectl get configmap ad-guid-migration -n cattle-system -o jsonpath='{.data.ad-guid-migration-status}{"\n"}'`
That command is meant to show the current status of the migration.

**Comment by anupama2501** [comment]
@crobby While the rancher server was up, I logged in as auth admin user and as a local admin, I tried getting the status of the configmap. 
```
kubectl get configmap ad-guid-migration -n cattle-system -o jsonpath='{.data.ad-guid-migration-status}{"\n"}'
Error from server (NotFound): configmaps "ad-guid-migration" not found
```
Since, the migration hasn't started the user was able to login but unable to list any of the resources. If I refresh the UI after a few minutes, I get logged out and then once I login as the auth admin user, I was able to load the resources properly.

Is there a way to restrict the users from logging in before the migration script starts?
